packages/bpk-component-chip-group/src/BpkChipGroup.tsx

```diff
@@ -0,0 +1,254 @@
+export type ChipGroupType = (typeof CHIP_GROUP_TYPES)[keyof typeof CHIP_GROUP_TYPES];
+export type ChipStyleType = (typeof CHIP_TYPES)[keyof typeof CHIP_TYPES];
+export type ChipComponentType = (typeof CHIP_COMPONENT)[keyof typeof CHIP_COMPONENT];
+
+export type SingleSelectChipItem = {
+  text: string;
+  accessibilityLabel?: string;
+  leadingAccessoryView?: ReactNode;
+  [rest: string]: any; // Inexact rest. See decisions/inexact-rest.md
+};
+
+export type ChipItem = {
+  component?: ChipComponentType;
+  onClick?: (selected: boolean, index: number) => void;
+  selected?: boolean;
+  hidden?: boolean;
+} & SingleSelectChipItem;
+
+export type InternalProps = {
+  chipGroupClassNames: string
+};
+
+export type CommonProps = {
+  label?: string;
+  ariaLabel?: string;
+  chipStyle?: ChipStyleType;
+  chips: ChipItem[];
+  ariaMultiselectable?: boolean;
+};
+
+export type RailChipGroupProps = {
+  stickyChip?: ChipItem;
+  leadingNudgerLabel: string;
+  trailingNudgerLabel: string;
+} & CommonProps;
+
+export type WrapChipGroupProps = {
+} & CommonProps;
+
+export type ChipGroupProps = (RailChipGroupProps & { type: typeof CHIP_GROUP_TYPES.rail } | WrapChipGroupProps & { type: typeof CHIP_GROUP_TYPES.wrap });
+
+const Chip = ({ ariaMultiselectable, chipIndex, chipItem, chipStyle }: { chipIndex: number, chipItem: ChipItem, chipStyle: ChipStyleType, ariaMultiselectable: boolean }) => {
+  const {
+    accessibilityLabel,
+    component = CHIP_COMPONENT.selectable,
+    hidden = false,
+    leadingAccessoryView = null,
+    onClick,
+    selected,
+    text,
+    ...rest
+  } = chipItem;
+  const Component = CHIP_COMPONENT_MAP[component];
+  return hidden ? null : (
+    <Component
+      key={text}
+      selected={selected ?? false}
+      type={chipStyle}
+      accessibilityLabel={accessibilityLabel || text}
+      onClick={() => {
+        if (onClick) {
+          onClick(!selected, chipIndex);
+        }
+      }}
+      role={ariaMultiselectable ? 'checkbox' : 'radio'}
+      leadingAccessoryView={leadingAccessoryView}
+      {...rest}
+    >
+      {text}
+    </Component>
+  );
+}
+
+const ChipGroupContent = (
+  { ariaLabel,
+    ariaMultiselectable,
+    chipGroupClassNames,
+    chipStyle,
+    chips,
+    label }:
+    {
+      chipGroupClassNames: string,
+      ariaMultiselectable: boolean,
+      ariaLabel?: string,
+      label?: string,
+      chips: ChipItem[],
+      chipStyle: ChipStyleType
+    }) => (
+  <fieldset
+    className={chipGroupClassNames}
+    role={ariaMultiselectable ? 'group' : 'radiogroup'}
+  >
+    {ariaLabel && <legend className='visually-hidden'>{ariaLabel}</legend>}
+    {label && <BpkText textStyle={TEXT_STYLES.footnote} aria-hidden>{label}</BpkText>}
+    {chips.map((chip, index) => <Chip chipItem={chip} chipStyle={chipStyle} ariaMultiselectable={ariaMultiselectable} chipIndex={index} />)}
+  </fieldset>
+);
+
+const RailChipGroup = ({
+  ariaLabel,
+  ariaMultiselectable = true,
+  chipGroupClassNames,
```

Comment:
What is the usecase for having a prop to provide additional classes? This would indicate that folks have to do customisation away from the design/its not fit for purpose that we want folks to change things

Reply:
I think we didn't want to duplicate this in both `RailChipGroup` and `WrapChipGroup`. Though I've removed the prop for now as I think it's less complex with the duiplication actually
```ts
  const chipGroupClassNames = getClassName(
    'bpk-chip-group',
    `bpk-chip-group--{type}`,
  );
```